File: leetcode/2863.cpp
#include "base.hpp"


class Solution {
public:
  int maxSubarrayLength(vector<int> &nums) {
    vector<int> picked;
    int n = nums.size();
    int minn_now = {INT_MAX};
    for (int i = n - 1; i >= 0; --i) {
      if (nums[i] < minn_now) {
        picked.push_back(i);
        minn_now = nums[i];
      }
    }
    reverse(picked.begin(),picked.end());
    int i{}, j{}, ans{1};
    for (; i < n; ++i) {
      while (j < picked.size() && (i>=picked[j] || nums[i]>nums[picked[j]])) {
        if(picked[j]-i+1 > ans){
          ans=picked[j]-i+1;
        }
        j++;
      }
      if(j==picked.size())break;
    }
    return ans==1?0:ans;
  }
};

int main() {}

File: leetcode/2863.py
from typing import List

class Solution:
    def maxSubarrayLength(self, nums: List[int]) -> int:
        nums = [(nums[i],i) for i in range(len(nums))]
        nums.sort(reverse = True)
        print(nums);
        prevminidx = float('inf')
        cur = float('-inf')
        curminidx = float('inf')
        res = 0
        for num, i in nums:
            if cur != num:
                prevminidx = min(prevminidx, curminidx)
                cur = num
            curminidx = i
            print(i,curminidx,prevminidx)
            res = max(res, i - prevminidx + 1)
        return res
      
s= Solution();
print(s.maxSubarrayLength([4,7,3,2,8]))

